src/sema/assembly/block.rs

```diff
@@ -79,7 +91,11 @@ pub(crate) fn process_statements(
             ns,
         ) {
             Ok(can_reach_next_statement) => {
-                if !reachable && !has_unreachable {
+                // There shouldn't be warnings of unreachable statements for function definitions
```

Comment:
I don't understand this comment. Could you elaborate the comment please?

Reply:
Take this example:
```
let x := foo(1, 2)
return(x, 2)
function foo(a, b) -> ret {
   ret := add(a, b)
}
```
Every statement after `return` is unreachable. Although the function definition is also a statement, it cannot be unreachable because it is not a statement that executes something. In addition, the function was invoked, so it is not dead code.

Reply:
Ok, that makes sense. I had not thought of that. Can you you add that to the comment in the code please?